examples/creatives/hover_nonlinear.js

```diff
@@ -0,0 +1,143 @@
+const QUARTER_THE_SIZE = .25;
+const THIRTY_FIVE_PERCENT = .35;
+const HALF_THE_SIZE = .5;
+const THREE_QUARTERS_THE_SIZE = .75;
+
+/* This creative expands and collapses when the user hovers over the banner. */
+class HoverNonLinear extends BaseSimidCreative {
+    constructor() {
+        super();
+
+        this.initialDimensions_ = null;
+    }
+
+    /** @override */
+    onInit(eventData) {
+        super.onInit(eventData);
+        this.initialDimensions_ = this.environmentData.creativeDimensions;
+        this.videoDimensionHeight = this.environmentData.videoDimensions.height;
+        this.videoDimensionWidth = this.environmentData.videoDimensions.width;
```

Comment:
you are declaring private instance variables here so i'd suggest creating them first in the constructor and updating them here as it isn't very clear that they persisit

Reply:
good idea